llvm/lib/Target/SPIRV/SPIRVInstructionSelector.cpp

```diff
@@ -1694,6 +1701,110 @@ bool SPIRVInstructionSelector::selectIntegerDot(Register ResVReg,
   return Result;
 }
 
+template <bool Signed>
+bool SPIRVInstructionSelector::selectDot4AddPacked(Register ResVReg,
+                                                   const SPIRVType *ResType,
+                                                   MachineInstr &I) const {
+  assert(I.getNumOperands() == 5);
+  assert(I.getOperand(2).isReg());
+  assert(I.getOperand(3).isReg());
+  assert(I.getOperand(4).isReg());
+  MachineBasicBlock &BB = *I.getParent();
+
+  Register Dot = MRI->createVirtualRegister(&SPIRV::IDRegClass);
+  bool Result = BuildMI(BB, I, I.getDebugLoc(), TII.get(SPIRV::OpDot))
```

Comment:
You want to use the [OpSDot](https://registry.khronos.org/SPIR-V/specs/unified1/SPIRV.html#OpSDot) and [OpUDot](https://registry.khronos.org/SPIR-V/specs/unified1/SPIRV.html#OpUDot) instructions. If I said OpDot before that was a mistake on my part.

Reply:
Updated. I had to define the ops in `SPIRVInstrInfo`, I noticed there seems to be a mirrored definition in the mlir  codebase (`mlir/Dialect/SPIRV/IR/SPIRVBase.td`). Is it required that I update that now as well?

Reply:
No you do not have to update the mlir code.